### PR TITLE
[Refactor][API2.0] Api ncnn openvino

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

This PR add `from_onnx` for ncnn and openvino

## Modification

requirement:
- [ ] https://github.com/open-mmlab/mmdeploy/pull/419

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/backend/ncnn/onnx2ncnn.py [hunk 1]
```diff
@@ -30,28 +27,3 @@ def get_output_model_file(onnx_path: str, work_dir: str) -> List[str]:
     save_param = osp.join(work_dir, file_name + '.param')
     save_bin = osp.join(work_dir, file_name + '.bin')
     return [save_param, save_bin]
-
-
-def onnx2ncnn(onnx_path: str, save_param: str, save_bin: str):
```
Comment (tpoisonooo) [inline mmdeploy/backend/ncnn/onnx2ncnn.py:35]: Why cut `onnx2ncnn.py` into 2 files ?

**Review by tpoisonooo — APPROVED** [review]
LGTM.

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

File: tools/deploy.py [hunk 2]
```diff
@@ -214,18 +214,17 @@ def main():
             logger.error('ncnn support is not available.')
             exit(1)
 
-        from mmdeploy.apis.ncnn import get_output_model_file, onnx2ncnn
+        import mmdeploy.apis.ncnn as ncnn_api
+        from mmdeploy.apis.ncnn import get_output_model_file
+
+        PIPELINE_MANAGER.set_log_level(logging.INFO, [ncnn_api.from_onnx])
```
Comment (lvhan028) [inline tools/deploy.py:220]: Why do we hard code `logging.INFO` in deploy.py? Can we use `args.log_level` ?

File: mmdeploy/backend/ncnn/onnx2ncnn.py [hunk 3]
```diff
@@ -32,26 +35,34 @@ def get_output_model_file(onnx_path: str, work_dir: str) -> List[str]:
...
+              output_file_prefix: str):
     """Convert ONNX to ncnn.
 
     The inputs of ncnn include a model file and a weight file. We need to use
     a executable program to convert the `.onnx` file to a `.param` file and
     a `.bin` file. The output files will save to work_dir.
 
     Example:
-        >>> from mmdeploy.backend.ncnn.onnx2ncnn import onnx2ncnn
+        >>> from mmdeploy.apis.ncnn import from_onnx
```
Comment (lvhan028) [inline mmdeploy/backend/ncnn/onnx2ncnn.py:47]: The example in docstring is supposed to be updated, as the arguments are changed.

File: mmdeploy/backend/openvino/onnx2openvino.py [hunk 4]
```diff
@@ -55,30 +56,33 @@ def get_output_model_file(onnx_path: str, work_dir: str) -> str:
...
 
     Examples:
-        >>> from mmdeploy.backend.openvino.onnx2openvino import onnx2openvino
+        >>> from mmdeploy.apis.openvino import from_onnx
         >>> input_info = {'input': [1,3,800,1344]}
         >>> output_names = ['dets', 'labels']
         >>> onnx_path = 'work_dir/end2end.onnx'
         >>> work_dir = 'work_dir'
-        >>> onnx2openvino(input_info, output_names, onnx_path, work_dir)
+        >>> from_onnx( onnx_path, work_dir, input_info, output_names)
```
Comment (lvhan028) [inline mmdeploy/backend/openvino/onnx2openvino.py:72]: The example in docstring is supposed to be updated, as the arguments are changed.

**Review by lvhan028 — APPROVED** [review]

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**